### PR TITLE
Add Dune section in OCaml content

### PR DESCRIPTION
**Pull request by Camilotk**
I would like to propose the addition of a tutorial on Dune, the OCaml build system, to the Fedora Developer Portal.

Dune is a fast and flexible build system for OCaml that has gained popularity among OCaml developers. It provides a uniform interface for building, testing, and installing OCaml projects, and helps manage dependencies between packages.

Adding a tutorial on Dune to the Fedora Developer Portal would be highly beneficial for OCaml developers who are new to the language and want to get started with building projects. The tutorial would provide a step-by-step guide on how to install Dune, start a new project, build the project, and run it, as well as how to add a library to a Dune project.

Inclusion of this tutorial would make the Fedora Developer Portal a more comprehensive resource for OCaml developers and help to attract new users to Fedora. I believe it would be a valuable addition to the existing resources on the portal.

I have included the tutorial in this pull request for your review. Please let me know if you have any questions or concerns. I look forward to your feedback.

Thank you for your time and consideration.

### CODE REVIEW

File: tech/languages/ocaml/dune.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+---
+title: Dune
+subsection: ocaml
+order: 2
+---
+
+# Dune
+
+Dune is a build system for OCaml. It provides a comprehensive and uniform interface for the building, testing, and installation of OCaml projects, making it incredibly easy for developers to manage dependencies between packages. Its primary focus is to offer simplicity, speed, and ease of use, making it a suitable solution for even the most complex and extensive OCaml projects. With Dune, you can enjoy the benefits of incremental builds, which result in faster build times by recompiling only the necessary files. Moreover, Dune makes it convenient to generate optimized native code and perform static analysis, contributing to the development of high-performance and reliable OCaml applications.
```
Comment (jackorp) [inline tech/languages/ocaml/dune.md:9]:
```suggestion
Dune is a build system for OCaml. It provides a comprehensive and uniform interface for the building, testing, and intalling OCaml projects, making it incredibly easy for developers to manage dependencies between packages. Its primary focus is to offer simplicity, speed, and ease of use, making it a suitable solution for even the most complex and extensive OCaml projects. With Dune, you can enjoy the benefits of incremental builds, which result in faster build times by recompiling only the necessary files. Moreover, Dune makes it convenient to generate optimized native code and perform static analysis, contributing to the development of high-performance and reliable OCaml applications.
```

File: tech/languages/ocaml/dune.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+## Install
+
+You can install Dune with OPAM that we installed in the enviroment setup. For do it just run:
```
Comment (jackorp) [inline tech/languages/ocaml/dune.md:13]:
```suggestion
You can install Dune with OPAM that we installed in the enviroment setup.

To install Dune using OPAM run:
```

Consider linking `environment setup` to the page

File: tech/languages/ocaml/dune.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+```console
+$ opam install dune
+```
+
+## Starting a new Project
+
+1. Create a new directory for your project.
```
Comment (jackorp) [inline tech/languages/ocaml/dune.md:21]: numbering in this whole section turns into all 1s.  I think the numbering has to be in succession if you want it numbered just like here: which renders correctly the 1 and 2 
```
## Building the project.
1...
2...
```

File: tech/languages/ocaml/dune.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+```console
+$ mkdir yourprojectname
+```
+
+2. Change into the new directory and run the following command to initialize a Dune project:
```
Comment (jackorp) [inline tech/languages/ocaml/dune.md:27]:
```suggestion
2. To initialize a Dune project change into the new directory and run the following command:
```

**Review by jackorp — CHANGES_REQUESTED** [review]
Some minor grammar fixes.
Also, please re-review whether you need to use the numbering. The markdown rendering messes up the correct number order unless the lines are in succession.

The number are probably not needed. I usually use normal paragraphs to discern steps that are in succession.

Content-wise this is great!

File: tech/languages/ocaml/dune.md (hunk 3 above)
Comment (Camilotk) [reply tech/languages/ocaml/dune.md]: Fixed https://github.com/developer-portal/content/pull/473/commits/7e02639e7f9c88ea9411215e7265a21d3926080a

**Comment by Camilotk** [comment]
All fixed.

**Comment by jackorp** [comment]
LGTM, thanks!

**Pull request merged**